model/src/cmd/ModelCommandAPI.cc

```diff
@@ -0,0 +1,437 @@
+
+        massInfo += "[" + std::to_string(massComp) + "]";
+        inertialInfo +=
+              "\n        [" + std::to_string(inertialMatrix.Ixx()) + " | "
+                            + std::to_string(inertialMatrix.Ixy()) + " | "
+                            + std::to_string(inertialMatrix.Ixz()) + "]\n"
+                "        [" + std::to_string(inertialMatrix.Ixy()) + " | "
+                            + std::to_string(inertialMatrix.Iyy()) + " | "
+                            + std::to_string(inertialMatrix.Iyz()) + "]\n"
+                "        [" + std::to_string(inertialMatrix.Ixz()) + " | "
+                            + std::to_string(inertialMatrix.Iyz()) + " | "
+                            + std::to_string(inertialMatrix.Izz()) + "]";
+        std::cout << "    - Mass: " << massInfo << std::endl
+                  << "    - Inertial Matrix: " << inertialInfo << std::endl;
+      }
+    }
+    {
+      auto poseComp =
+          ecm.Component<ignition::gazebo::components::Pose>(_entity);
+
+      std::string poseInfo;
+      if (poseComp)
+      {
+        poseInfo +=
+              "\n        [" + std::to_string(poseComp->Data().X()) + " | "
+                            + std::to_string(poseComp->Data().Y()) + " | "
+                            + std::to_string(poseComp->Data().Z()) + "]\n"
+                "        [" + std::to_string(poseComp->Data().Roll()) + " | "
+                            + std::to_string(poseComp->Data().Pitch()) + " | "
+                            + std::to_string(poseComp->Data().Yaw()) + "]";
+
+        std::cout << "    - Pose: " << poseInfo << std::endl;
+      }
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+void printJoints(uint64_t entity,
+                ignition::gazebo::EntityComponentManager &ecm,
+                std::string joint_name)
+{
+  auto joints = ecm.EntitiesByComponents(
+  ignition::gazebo::components::ParentEntity(entity),
+  ignition::gazebo::components::Joint());
+
+  for(auto &_entity : joints)
+  {
+    {
+      auto parentComp =
+          ecm.Component<ignition::gazebo::components::ParentEntity>(_entity);
+
+      auto nameComp =
+          ecm.Component<ignition::gazebo::components::Name>(_entity);
+
+      if(joint_name.length() && joint_name != nameComp->Data())
+          continue;
+
+      std::string parentInfo;
+      if (parentComp)
+      {
+        auto parentNameComp =
+            ecm.Component<ignition::gazebo::components::Name>(
+            parentComp->Data());
+
+        if (parentNameComp)
+        {
+          parentInfo += parentNameComp->Data() + " ";
+        }
+        parentInfo += "[" + std::to_string(parentComp->Data()) + "]";
+      }
+
+      std::cout << "  - Joint [" << _entity << "]" << std::endl
+                << "    - Name: " << nameComp->Data() << std::endl
+                << "    - Parent: " << parentInfo << std::endl;
+    }
+    {
+
+      std::map<sdf::JointType, std::string> jointTypes = {
```

Comment:
You can make this map `static const `.
Move it to the beginning of the method.

Reply:
Addressed